server/controllers/user.js

```diff
@@ -0,0 +1,195 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt-nodejs');
+
+const secretKey = process.env.SECRET_TOKEN_KEY;
+
+const saltRounds = bcrypt.genSaltSync(10);
+
+const User = require('../models').User;
+const document = require('../models').Document;
+
+// create an static method to handle inclutions
+
+const findWithDocuments = (method, params) => {
+  const includeDocuments = {
+    include: [{
+      model: document,
+      as: 'documents',
+    }],
+  };
+
+  if (params) {
+    return User[method](params, includeDocuments);
+  }
+
+  return User[method](includeDocuments);
+};
+
+module.exports = {
+
+  // create a new user
+
+  create(req, res) {
+    const firstName = req.body.firstName;
+    const lastName = req.body.lastName;
+    const userName = req.body.userName;
+    const email = req.body.email;
+    const password = req.body.password;
+    const roleId = 2;
+
+    if (!firstName || !lastName || !userName || !email || !password) {
+      return res.status(400).json({ message: 'Enter All Required Fields' });
+    } else if (!(/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(req.body.email))) {
```

Comment:
minor. For readability, the regex string could be a constant at the beginning of the file. So it can be reused both here, and during login, if login takes email.